lib/Dialect/Handshake/HandshakeTypes.cpp

```diff
@@ -99,25 +92,15 @@ checkChannelExtra(function_ref<InFlightDiagnostic()> emitError,
   return success();
 }
 
-static Type parseChannelAfterLess(AsmParser &odsParser) {
-  FailureOr<Type> dataType;
+/// Parse the extra signals from the IR representation.
+/// For example: [spec: i1, tag: i32]
+/// This function is common to both ControlType and ChannelType.
+/// TODO: Is this proper to use std::optional for error handling?
+static std::optional<SmallVector<ExtraSignal>>
```

Comment:
I wanted to handle the error case separately from the empty vector case, so the result type became somewhat unconventional. If you have a good idea, please let me know!